thread_with_timeout/thread_with_timeout.cpp

```diff
@@ -149,28 +151,41 @@ class BaseStreamSession {
   }
   void Start(BaseStreamSession* session, int timeout,
              SATRT_COMPLETE_FUNC funt_ptr) {
-    if (state_ != kInited) return;
+    if (state_ != kInited) {
+      if (state_ == kStarting) {
+        std::cout << "already starting...\n";
+      }
+      return;
+    }
     state_ = kStarting;
-    std::thread t =
-        std::thread(&BaseStreamSession::RunStartNew, this, funt_ptr, timeout);
-    if (t.joinable())
-    {
+    api_thread_ = std::thread(&BaseStreamSession::RunStartNew, this, funt_ptr, timeout);
+    return;
+  }
+  void Stop(BaseStreamSession* session, int timeout,
+      SATRT_COMPLETE_FUNC funt_ptr) {
+    if (state_ != KReady) return;
+    state_ = kStopping;
+    std::thread t = std::move(api_thread_);
```

Comment:
should i just use    
 if (api_thread_.joinable()) {
      api_thread_.join();
    }

Reply:
yes